src/Compilers/Core/Portable/StreamExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.IO;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static class StreamExtensions
+    {
+        /// <summary>
+        /// Attempts to read all of the requested bytes from the stream into the buffer
+        /// </summary>
+        /// <returns>
+        /// The number of bytes read. Less than <paramref name="count" /> will
+        /// only be returned if the end of stream is reached before all bytes can be read.
+        /// </returns>
+        public static int TryReadAll(
```

Comment:
Generally methods prefixed with Try return a `bool`.  Trying to think of a better name / pattern here though and not coming up with anything great.